src/group/mls_group/apply_commit.rs

```diff
@@ -90,11 +84,14 @@ impl MlsGroup {
             }
             if is_own_commit {
                 // Find the right KeyPackageBundle among the pending bundles
-                let own_kpb = pending_kpbs
+                let own_kpb_index = match pending_kpbs
                     .iter()
-                    .find(|&kpb| kpb.get_key_package() == kp)
-                    .unwrap()
-                    .clone();
+                    .position(|kpb| kpb.get_key_package() == kp)
+                {
+                    Some(i) => i,
+                    None => return Err(ApplyCommitError::MissingOwnKeyPackage),
+                };
+                let own_kpb = pending_kpbs.remove(own_kpb_index);
```

Comment:
I'm not sure we need to remove keys from the pending key packages, because we should drop all of them anyway on epoch changes. The rule is that proposals are not carried over between epochs and therefore I don't see a reason to keep the private keys around.
It has nothing to do with your changes directly, but I thinks this is the right PR to make that change, can you add that at the end of the function?

Reply:
We have to remove the pending key here because it is consumed. We didn't do this before because we were able to clone them. Now we have to cleanly handle keys without cloning.
This ensure that we can't re-use them. At the same time does this function consume the vector of key packages such that they can't be used later and there's no need to explicitly remove any.

Reply:
Agreed, this makes sense. We should still drop all of them at the end of the epoch. It could be that not all keys were consumed, because not all updates are covered by the Commit, those keys should be dropped as well.

Reply:
I filed #119 to follow up on this.